src/LDNS.xs

```diff
@@ -2543,6 +2542,40 @@ rr_nsec3_covers(obj,name)
     OUTPUT:
         RETVAL
 
+SV *
+rr_nsec3_hash_name(obj,name)
+    Zonemaster::LDNS::RR::NSEC3 obj;
+    const char *name;
+    INIT:
+        /* Sanity test on owner name */
+        if (ldns_dname_label_count(ldns_rr_owner(obj)) == 0)
+            XSRETURN_UNDEF;
+    CODE:
+    {
+        ldns_rr *clone;
+        ldns_rdf *hashed;
+        ldns_rdf *dname;
+
+        dname = ldns_rdf_new_frm_str(LDNS_RDF_TYPE_DNAME, name);
+        if (!dname)
+            XSRETURN_UNDEF;
+
+        ldns_dname2canonical(dname);
+
+        clone = ldns_rr_clone(obj);
+        ldns_rr2canonical(clone);
```

Comment:
Is it necessary to clone that object? Is it necessary to canonicalize it? `ldns_nsec3_hash_name_frm_nsec3` doesn’t modify its first argument (the NSEC3 resource record) and only uses the algorithm, salt and iterations from it.